Patch-Source: https://github.com/Qalculate/libqalculate/pull/386
From dbe890a141e5f962eecfb655efffaa02db7d4a51 Mon Sep 17 00:00:00 2001
From: Jakub Jirutka <jakub@jirutka.cz>
Date: Fri, 21 Jan 2022 00:15:13 +0100
Subject: [PATCH] Do not rewrite -Os to -O2

qalc doesn't segfault when compiled with -Os (gcc 11.2.1), and if it
does, that means you have an undefined behaviour here which may result
in security vulnerability.
---
 configure.ac | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/configure.ac b/configure.ac
index a407c365..722413ad 100644
--- a/configure.ac
+++ b/configure.ac
@@ -51,21 +51,6 @@ dnl  ------------------------------------
 dnl | check for compiler characteristics |-------------------------------------
 dnl  ------------------------------------
 
-dnl Replace -Os with -O2 to stop segfault on startup
-if test "x$GCC" = "xyes"; then
-	case $CFLAGS in
-	*-Os*)
-		CFLAGS="$CFLAGS -O2"
-		;;
-	esac
-	case $CXXFLAGS in
-	*-Os*)
-		CXXFLAGS="$CXXFLAGS -O2"
-		;;
-	esac
-fi
-
-
 dnl Use -Wall if we have gcc.
 changequote(,)dnl
 if test "x$GCC" = "xyes"; then
